compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -1983,7 +1991,10 @@ private STNode parseComplexTypeDescriptor(STNode typeDesc, ParserRuleContext con
                 return parseComplexTypeDescriptor(parseOptionalTypeDescriptor(typeDesc), context);
             // If next token after a type descriptor is <code>[</code> then it is an array type descriptor
             case OPEN_BRACKET_TOKEN:
-                return parseComplexTypeDescriptor(parseArrayTypeDescriptor(typeDesc), context);
+                if (isListBindingPattern()) { // return typed binding attern if we find a list-b-p
+                    return STNodeFactory.createTypedBindingPatternNode(typeDesc, parseListBindingPattern());
+                }
```

Comment:
It is not correct to create typed-binding-patterns here. Reason is, this is a generic method to parse type-descs. 

But typed-binding-patterns are not supported for all places where a type is supported.

e.g: type-cast, func-return, type-parameter, etc. there are many places where a type-desc can appear without a binding pattern.

Reply:
hmmm, that makes sense.